transcom/mymove
# B-21363 INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21363)
## [Previous B-21363 PR (combined with B-21362)](https://github.com/transcom/mymove/pull/13837)

## Summary

This is adding in some explicit feature flag checks for UB shipments, dependent on the unaccompanied_baggage FF being enabled or not.

To test, please verify the following AC:

Use the createMTOShipment to create a UB. Example body for creating a shipment with the createMTOShipment:
```
{
  "moveTaskOrderID": "91154a53-6880-460b-a270-57bcb53d631a",
  "shipmentType": "UNACCOMPANIED_BAGGAGE",
  "requestedPickupDate": "2024-10-04",
  "pickupAddress": {
      "streetAddress1": "123 Main St",
      "city": "Nicholasville",
      "state": "KY",
      "postalCode": "40356",
      "country": "US",
      "county": "JESSAMINE"
   }
}
```

Given I am interacting with the MilMove customer app
When the unaccompanied_baggage FF is disabled and I try to create a shipment
Then I receive a 422 response with a message of "Unaccompanied baggage shipments can't be created unless the unaccompanied_baggage feature flag is enabled"

Given I am interacting with the MilMove customer app
When I have previously created a UB shipment with the unaccompanied_baggage FF on, but then turn the unaccompanied_baggage FF off
Then I can no longer see the UB shipments listed

With the UB FF on after creating UB shipments:
![image](https://github.com/user-attachments/assets/dfd4bbec-f548-425b-92e0-f8c5906182b0)

With the UB FF off after creating UB shipments:
![image](https://github.com/user-attachments/assets/da0c8525-c4fd-4511-a908-a7641f7f2828)
